src/ipc/TypedMsgHdr.cc

```diff
@@ -18,25 +18,40 @@
 
 Ipc::TypedMsgHdr::TypedMsgHdr()
 {
-    memset(this, 0, sizeof(*this));
+    clear();
     sync();
 }
 
 Ipc::TypedMsgHdr::TypedMsgHdr(const TypedMsgHdr &tmh)
 {
-    memcpy(this, &tmh, sizeof(*this));
-    sync();
+    operator =(tmh);
```

Comment:
It was not doing so before and I was reluctant to add new behaviour here since I am still a bit sketchy on which operators get called for the placement-new sequence for these SMP sharing objects.

Added. I trust that you know better than me on the above.

Reply:
> I was reluctant to add new behaviour here since I am still a bit sketchy on which operators get called for the placement-new sequence for these SMP sharing objects.

Just FYI:

* I am OK with the latest version of the copy constructor.
* I was not asking to add new behavior, just a safe(r) replacement of the old one.
* IIRC, Ipc::TypedMsgHdr objects are not allocated in shared memory.
* Both placement `new` and regular `new` result in the same constructor call. Constructors do not allocate memory. They are called after the memory is allocated (or otherwise provided). And placement `new` does not trigger some special constructor call.